hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -236,6 +256,10 @@ public boolean equals(Object o) {
             return false;
         }
 
+        if (type == IndexType.BITMAP && !bitmapIndexOptions.equals(that.bitmapIndexOptions)) {
```

Comment:
This seems wrong to me because different objects may yield `true`. Same for `hashCode`.

Reply:
Yes, but bitmap index options are in some sense irrelevant to other index types. For instance, two `HASH` index definitions having different bitmap index options may result in the same hash index being created.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/16305/commits/123a22619c1c2ed7d26cabc286f88dbe801cff06